lib/rules/object-shorthand.js

```diff
@@ -129,19 +129,8 @@ module.exports = {
                     return;
                 }
 
-                // if we're "never" and concise we should warn now
-                if (APPLY_NEVER && isConciseProperty) {
-                    type = node.method ? "method" : "property";
-                    context.report(node, "Expected longform " + type + " syntax.");
-                }
-
-                // {'xyz'() {}} should be written as {'xyz': function() {}}
-                if (AVOID_QUOTES && isStringLiteral(node.key) && isConciseProperty) {
-                    context.report(node, "Expected longform method syntax for string literal keys.");
-                }
-
-                // at this point if we're concise or if we're "never" we can leave
-                if (APPLY_NEVER || AVOID_QUOTES || isConciseProperty) {
```

Comment:
The cause was this `AVOID_QUOTES`.
This has been replaced by L166-168 & L177-179.